### PR TITLE
upgrades changeling disguises

### PR DESCRIPTION
**Pull request by theOperand**
they now get nodrop flesh clothing stuff which should hopefully make disguising themselves a little easier.

### CODE REVIEW

File: code/game/gamemodes/changeling/powers/transform.dm [hunk 1]
```diff
@@ -7,17 +7,138 @@
...
+//changeling clothing items
+/obj/item/clothing/head/changeling
+	name = "flesh"
+	flags = NODROP
+
+/obj/item/clothing/gloves/changeling
+	name = "flesh"
+	flags = NODROP
+
+/obj/item/clothing/head/changeling
```
Comment (RemieRichards) [inline code/game/gamemodes/changeling/powers/transform.dm]: defined twice nerd.

Comment (Miauw62) [inline code/game/gamemodes/changeling/powers/transform.dm]: whoops

File: code/game/gamemodes/changeling/powers/transform.dm [hunk 2]
```diff
@@ -7,17 +7,138 @@
...
 		return
 
-	user.dna = chosen_dna
-	user.real_name = chosen_dna.real_name
+	user.dna = chosen_prof.dna
+	user.real_name = chosen_prof.name
 	hardset_dna(user, null, null, null, null, chosen_dna.species.type, chosen_dna.features)
+	var/list/slots = list("head", "wear_mask", "back", "wear_suit", "w_uniform", "shoes", "belt", "gloves", "glasses", "ears", "wear_id", "s_store")
+
+	//im so sorry
```
Comment (RemieRichards) [inline code/game/gamemodes/changeling/powers/transform.dm]: Jesus, cant you loop through get_inventory_slots() or another similar inventory proc?

Comment (Miauw62) [inline code/game/gamemodes/changeling/powers/transform.dm]: i need special paths for most slots.

i think i can solve that with an associative list though???

Comment (RemieRichards) [inline code/game/gamemodes/changeling/powers/transform.dm]: why do you need separate paths, just change the slot id on the default flesh clothing.

File: code/game/gamemodes/changeling/powers/transform.dm [hunk 3]
```diff
@@ -7,17 +7,138 @@
 	req_human = 1
 	max_genetic_damage = 3
 
+//changeling clothing items
+/obj/item/clothing/head/changeling
```
Comment (RemieRichards) [inline code/game/gamemodes/changeling/powers/transform.dm]: I've decreed that this is garbage, please have only ONE flesh clothing type, and change the Slot ID on it as needed.

Comment (Miauw62) [inline code/game/gamemodes/changeling/powers/transform.dm]: are you sure that having it be one type won't cause issues when things try to read variables that arent there?

Comment (RemieRichards) [inline code/game/gamemodes/changeling/powers/transform.dm]: I'm fairly certain most of the inventory slots are only defined as /item, meaning any coded assuming they're NOT /item, are already in the wrong.

Comment (RemieRichards) [inline code/game/gamemodes/changeling/powers/transform.dm]: Fucking, Masks are defined as actual masks, most other slots are /item. looks like it isn't possible (unless you repath mask code to not assume type, like all other slots)

Comment (Dorsisdwarf) [inline code/game/gamemodes/changeling/powers/transform.dm]: Possible solution: use generic "flesh" for all slots except mask, use mask "flesh" item for the mask slot?

**Comment by Cheridan** [comment]
it's been like 2 years you can just call them "changelings" now

**Comment by Miauw62** [comment]
alright it works now except hair and undershirts dont because ????

PLS HELP

**Comment by Miauw62** [comment]
alright here we go the new transform now works properly.
feel free to try it out.

**Comment by Miauw62** [comment]
conflicts fixed, changelog added, ready 4 merge.

File: code/game/gamemodes/changeling/changeling.dm [hunk 4]
```diff
@@ -277,18 +281,19 @@ var/list/possible_changeling_IDs = list("Alpha","Beta","Gamma","Delta","Epsilon"
...
 			return 1
 	return 0
 
-/datum/changeling/proc/can_absorb_dna(mob/living/carbon/user, mob/living/carbon/target)
-	if(absorbed_dna[1] == user.dna)//If our current DNA is the stalest, we gotta ditch it.
+/datum/changeling/proc/can_absorb_dna(mob/living/carbon/user, mob/living/carbon/human/target)
+	var/datum/changelingprofile/prof = stored_profiles[1]
+	if(prof.dna == user.dna)//If our current DNA is the stalest, we gotta ditch it.
 		user << "<span class='warning'>We have reached our capacity to store genetic information! We must transform before absorbing more.</span>"
 		return
```
Comment (phil235) [inline code/game/gamemodes/changeling/changeling.dm]: This check is true for lings that haven't absorbed any dna yet (and maybe other cases). This means you can never absorb dna and always get "We have reached our capacity to store genetic information! " with every form of dna absorption.

**Comment by Miauw62** [comment]
Should be fixed.

**Comment by Miauw62** [comment]
this should fix it

**Comment by Miauw62** [comment]
should be ready to merge

**Pull request merged**